### ISSUE TITLE
Purchasing the Commerce plan and DIFM Express breaks the connection to DIFM Express tools

### ISSUE THREAD

**Issue by dcoleonline** [Automattic/wp-calypso#94517]
### Quick summary

There is currently an issue that occurs when a customer purchases DIFM Express with the Commerce plan. The Commerce plan pushes the site AT, breaking the connection needed for our content submission form and other DIFM Express automation tools.  Discussed here p58i-icz-p2#comment-64349

### Steps to reproduce

**1. Add $1,039 free credits to your WP account**

**2. Add a new site with the Commerce plan to your cart**
* Go to https://wordpress.com/sites and click “Add new site”
* Add a free domain and click “Continue”
* Select the “Get Commerce” to add the Commerce plan to your cart

**3. Add DIFM Express to the cart**
* Browse to the DIFM Express onboarding for stores https://wordpress.com/start/do-it-for-me/new-or-existing-site?ref=supportpage
* Select “Use existing site”
* Select the site with the free domain you used for the Commerce plan
* Go through the onboarding flow to get to the cart

**4. Checkout using your free credits**
![difm-express-checkout](https://github.com/user-attachments/assets/b880c875-08db-4f53-9e94-88cb5af235f1)

**5. If you see the content form after checkout, try submitting content. You will see an error and will not be able to submit content** 
![content-form-error](https://github.com/user-attachments/assets/c3d399f0-5118-4d2a-9982-7ef6fe4cd8d1)

**If you paste the URL in a new tab, we see this error:**
![direct-link](https://github.com/user-attachments/assets/239138e1-e28d-447b-9d2a-5f24ed520736)

**6. Check the DIFM tools in the Blog RC**
* Go to the Blog RC for the new site
* Scroll down to the DIFM Tools section and see the note `No Built By Express options found.’ 
![DIFM-tools](https://github.com/user-attachments/assets/69f21444-5bd4-429a-9ad0-210e24ddf659)
* Connection is broken.

**7. Review the audit trail to see when the connection breaks in the sequence** 
![blog-audit](https://github.com/user-attachments/assets/5880b402-49f3-4537-bead-aa3e21c5fe1d)

### What you expected to happen

The site should stay connected to the DIFM Tools and the form should work for the customer to submit content.

When the connection is working, we should see the link to their content form under DIFM Tools like this:
![working-tools](https://github.com/user-attachments/assets/d0b81167-1c8e-4001-842b-47bfbe0f2174)

### What actually happened

The Commerce plan pushes the site atomic right after purchase, which breaks the connection to the DIFM Tools. The customer is not able to submit their content in the form.

### Impact

Some (< 50%)

### Available workarounds?

Yes, difficult to implement

### If the above answer is "Yes...", outline the workaround.

We have a time-consuming workaround outlined here → Process: Fixing Content Form on AT Sites (Pdh1Xd-2Ni-p2)

That being said, there are plans (p58i-icz-p2#comment-64349) to start pushing all new WP.com Business plan sites atomic after purchase. If this happens, this form issue will start to greatly affect the efficiency of our team.

**Comment by retnonindya** [comment]
📌 REPRODUCTION RESULTS
Replicated

📌 FINDINGS/SCREENSHOTS/VIDEO
Not only user can't submit content, I also found the warning confusing:

<img width="1686" alt="Screenshot 2024-09-18 at 4 27 30 PM" src="https://github.com/user-attachments/assets/8a2c5d16-dc2b-45e1-b0a4-5112a0b6f005">

The warning about "please check your Internet connection" implies that the issue is on the user's Internet connection, while it's more on website/Jetpack connection.

📌 ACTIONS
- Triaged
- I noticed we had informed folks here: p58i-icz-p2#comment-64349
- Considering the severity of the issue, I'm raising this issue's priority to High

Thanks heaps for reporting this, @dcoleonline!

**Comment by candy02058912** [comment]
@Automattic/quake will be moving this project forward: p58i-icz-p2#comment-64402

**Comment by p-jackson** [comment]
The same problem existed when the site was originally created using the DIFM flow, which is now fixed by D162026-code.
Still working on the case described in this issue's steps to reproduced, where the DIFM flow is used to add the DIFM product to an existing site. It's a bit trickier.

**Comment by autumnfjeld** [comment]
Discussed here: https://dotcom.wordpress.com/2024/09/12/idea-default-new-business-sites-to-atomic-wp-admin/#comment-64349

**Comment by p-jackson** [comment]
The same issue exists with a new flow I discovered: `do-it-for-me-store`
Fix: https://code.a8c.com/D162188

**Comment by paulopmt1** [comment]
@autumnfjeld Although we're pausing the project, fixing the root issue seems to be a good maintenance task to prioritize. Thoughts?

**Comment by autumnfjeld** [comment]
Yes, I agree. It's a good maintenance task and it would be great if we continue work and fix this issue.

**Comment by obenland** [comment]
> It's a good maintenance task

That doesn't sound like this should be a Blocker level issue then.

@retnonindya I'll lower the priority on this. Blockers should be something that needs our immediate attention and be solved as quickly as possible.

**Comment by donnapep** [comment]
Hi @p-jackson. Just checking in on this high priority issue to see if there's anything new to report?

**Comment by davemart-in** [comment]
Removed from The One Board, since Quake picked it up.

**Comment by davemart-in** [comment]
That's a good question. From my perspective:

A) If someone grabs an issue, but doesn't assign it to their teams board, it's okay to stay in The One Board.
B) But once someone assigns it to a team board, it makes sense to remove from The One Board, else as people are going through The One Board, they then have to sift through a bunch of issues that already have a home. That sifting process can be taxing. 

If I'm doing this wrong though, I'm happy to change. cc @inaikem

**Comment by p-jackson** [comment]
> Hi @p-jackson. Just checking in on this high priority issue to see if there's anything new to report?

@donnapep no news yet sorry

**Comment by donnapep** [comment]
> Shouldn't it stay on the One Board for tracking purposes

We should go ahead and remove issues from The One Board "if they're being accounted for in other ways". Reference - p1728657406004539-slack-C07GZ2UA3TN

**Comment by autumnfjeld** [comment]
@p-jackson  could you add a quick update to this issue on work still remaining?  WOuld be great to have a "Done" and "Next" section to make it clear. 

I think this PR was the most recent that contributed to this work. 
https://github.com/Automattic/wp-calypso/pull/95183

**Comment by daledupreez** [comment]
I've started working on a fix in 174382-ghe-Automattic/wpcom.

**Comment by daledupreez** [comment]
Noting that I deployed 174382-ghe-Automattic/wpcom a little while ago, but we still need https://github.com/Automattic/wp-calypso/pull/100536 deployed to get this fully resolved.

**Comment by daledupreez** [comment]
@aneeshd16, after your deploy of #100536 yesterday, do you think we can close this issue?

**Comment by aneeshd16** [comment]
@daledupreez, I think so! @dcoleonline, can you please confirm if the issue has been resolved and we can close this?

**Comment by jartes** [comment]
Closing as I can no longer replicate. Thank you!
